src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -129,6 +133,45 @@ public void updateFilteredContactList(Predicate<Contact> predicate) {
         filteredContacts.setPredicate(predicate);
     }
 
+    //=========== Module ================================================================================
+
+    @Override
+    public boolean hasModule(Module module) {
+        requireNonNull(module);
+        return trackIter.hasModule(module);
+    }
+
+    @Override
+    public void deleteModule(Module target) {
+        trackIter.removeModule(target);
+    }
+
+    @Override
+    public void addModule(Module module) {
+        trackIter.addModule(module);
+        updateFilteredModuleList(PREDICATE_SHOW_ALL_MODULES);
+    }
+
+    @Override
+    public void setModule(Module target, Module editedModule) {
+        requireAllNonNull(target, editedModule);
+
+        trackIter.setModule(target, editedModule);
+    }
```

Comment:
Can I check, what is the difference between setModule and setModules? They are for different use cases?

Reply:
Yeah, since originally there were setContact and setContacts as well